Remove auth mode (#32)
ietf-wg-jose/draft-ietf-jose-hpke-encrypt: draft-ietf-jose-hpke-encrypt.md

HPKE supports several modes, which are described in Table 1 of {{RFC9180}}.

> In JOSE-HPKE, only the "base_mode" is used.

Author: ```suggestion
In JOSE-HPKE, only the "base_mode", "mode_psk" is used.
```

> In JOSE-HPKE, the "base_mode" and "mode_psk" are determined by the presence of the JOSE Header parameter "psk_id"

Author: ```suggestion
In JOSE-HPKE, only "mode_base" and "mode_psk" are supported.
When "psk_id" JOSE Header parameter is present the mode is "mode_psk", otherwise the mode is "mode_base".
```

- The protected header MUST contain an "alg" that is JOSE-HPKE algorithm.

> - The protected header MUST contain an "enc" with value "dir". This is an explicit exception to requirement in Section 4.1.2 of {{RFC7516}} that
> "enc" must be an AEAD algorithm. This is appropriate, as HPKE will perform plaintext encryption.

Author: ```suggestion
- The protected header MUST contain an "enc" with value "dir". This is an explicit exception to requirement in Section 4.1.2 of {{RFC7516}} that
"enc" must be an AEAD algorithm. This is appropriate, as HPKE will perform plaintext encryption.
- The protected header parameters "psk_id" MAY be present.
```

- When all recipients use the same HPKE algorithm to secure the Content Encryption Key, the JWE Protected Header SHOULD contain "alg".
Otherwise, the JWE Protected Header (and JWE Shared Unprotected Header) MUST NOT contain "alg".

> - JOSE Header parameter "alg" MUST be a JOSE-HPKE algorithm.

Author: ```suggestion
- JOSE Header parameter "alg" MUST be a JOSE-HPKE algorithm.
- JOSE Header parameter "psk_id" MAY be present.
```

This is guidance is relevant to both Key Encryption and Integrated Encryption.
When using Key Encryption, the strength of the content encryption algorithm should not be significantly different from the strengh of the Key Encryption algorithms used.

> ## Authenticated KEM
> 
> Authenticated KEM is not supported in JOSE HPKE for the following reasons:
> 
> 1. Authenticated KEMs provide authentication at the KEM level, which has different implications depending on whether they are used for key
> encryption or integrated encryption. In integrated encryption, the KEM operations directly secure the message plaintext. In key encryption,
> the KEM operations secure the content encryption key. Using authenticated KEMs for key encryption does not provide meaningful security
> benefits and will create a false sense of data origin authentication.
> 
> 2. The PQ/T Hybrid KEM in HPKE is not an authenticated KEM and can only be used with HPKE Base mode. Similarly, PQ KEMs in HPKE are not
> authenticated KEMs, making them incompatible with an authenticated KEM approach.
> 

Author: ```suggestion
## HPKE Mode Support

TODO: Summarize that only base and psk mode are supported
```

This is guidance is relevant to both Key Encryption and Integrated Encryption.
When using Key Encryption, the strength of the content encryption algorithm should not be significantly different from the strengh of the Key Encryption algorithms used.

> ## HPKE Mode Support
> 
> TODO: Summarize that only base and psk mode are supported

Author: ```suggestion
```

- Change Controller: IETF
- Specification Document(s):   RFCXXXX

> 

Author: ```suggestion
### psk_id

- Header Parameter Name: "psk_id"
- Header Parameter Description: A key identifier (kid) for the pre-shared key as defined in { Section 5.1.2 of RFC9180 }
- Header Parameter Usage Location(s): JWE
- Change Controller: IETF
- Specification Document(s):   RFCXXXX
```
